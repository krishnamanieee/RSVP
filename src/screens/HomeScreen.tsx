import React from 'react';
import {Text, View} from 'react-native';

class HomeScreen extends React.PureComponent<any, any> {
  render(): React.ReactElement {
    return (
      <View>
        <Text>Demo</Text>
      </View>
    );
  }
}
export default HomeScreen;
