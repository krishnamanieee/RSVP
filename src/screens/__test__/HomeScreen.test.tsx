import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import HomeScreen, {IHomeScreenProps, IHomeScreenState} from '../HomeScreen';

jest.mock('react-native-gesture-handler');

const testProps = (Props: IHomeScreenProps) => ({
  ...Props,
});

const mockFunction = jest.fn();

let props: IHomeScreenProps = {
  navigation: mockFunction,
};

describe('HomeScreen', () => {
  let wrapper: any;
  let instance: HomeScreen;
  beforeEach(() => {
    wrapper = shallow<HomeScreen, IHomeScreenProps, IHomeScreenState>(
      <HomeScreen {...props} />,
    );
    instance = wrapper.instance();
  });

  it('should match match snapshot', () => {
    const tree = renderer.create(<HomeScreen {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
