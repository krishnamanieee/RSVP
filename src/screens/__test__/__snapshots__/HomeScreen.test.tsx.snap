// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "paddingVertical": 5,
        "textAlign": "center",
      }
    }
  >
    Note: Please do pull to refresh to update the list
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Search the participant"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
